gridsingularity/gsy-e
# D3ASIM-3626: Add flag for Degrees of Freedom

Add a flag to enable/disable Degrees of Freedom.

Note: we define as DoF the `attributes` and `requirements` arguments of the orders.

## Steps

To test the changes you can do the following
1. Modify `test_redis_aggregator` setup in d3a-api-client by adding DoF to the sent orders. For example:
    ```python
    self.add_to_batch_commands.bid_energy(
        area_uuid=area_uuid, price=30, energy=energy, 
        attributes={"some-attribute": 12})
    ```

2. Run d3a and d3a-api-client:
    ```shell
    d3a run --setup strategy_tests.external_devices --no-export --seed 0 --enable-external-connection --disable-dof --paused
    d3a-api-client -l INFO run --setup test_redis_aggregator
    ```

You should now be able to see that the DoF have been correctly stripped off the orders. To do that, you can check the returned output of the batch aggregator command in d3a-api-client. It should contain a message like the following:
> {..., 'message': "The following arguments are not supported for this market and have been removed from your order: ['attributes']."}
